annotation-solr/src/main/java/eu/europeana/annotation/solr/service/impl/SolrAnnotationServiceImpl.java

```diff
@@ -563,17 +564,25 @@ private void addNotSelfDupplicateFilter(Annotation anno, SolrQuery query, boolea
 	
 	private SolrQuery solrUniquenessQuerySubtitles(Annotation anno, boolean noSelfDupplicate) {
       SolrQuery query = new SolrQuery();
-      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getSource() + "\"");
-      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.SUBTITLING.getOaType() + "\"");
-      query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE_PREFIX + anno.getBody().getLanguage() + ":*");
+      query.setQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.SUBTITLING.getOaType() + "\"");
+      query.addFilterQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getSource() + "\"");
+      String bodyText = extractTextValues(anno.getBody());
+      query.addFilterQuery(SolrAnnotationConstants.BODY_VALUE_PREFIX + anno.getBody().getLanguage() + ":\"" + bodyText + "\"");
       addNotSelfDupplicateFilter(anno, query, noSelfDupplicate);
       return query;
 	}
 	
 	private SolrQuery solrUniquenessQuerySemanticTagging(Annotation anno, boolean noSelfDupplicate) {
       SolrQuery query = new SolrQuery();
-      query.setQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + anno.getTarget().getValue() + "\"");
-      query.addFilterQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      query.setQuery(WebAnnotationModelFields.MOTIVATION + ":\"" + MotivationTypes.TAGGING.getOaType() + "\"");
+      if(anno.getTarget()!=null) {
+        List<String> targetValues = extractUriValues(anno.getTarget());
+        if(targetValues!=null) {
+          for(String target: targetValues) {
+            query.addFilterQuery(SolrAnnotationConstants.TARGET_URI + ":\"" + target + "\"");
```

Comment:
we should probably switch to use the target_record_uri, still need to agree this with Hugo

Reply:
Only for semantic tagging and LinkForContributions, (meaning the cases where we don't check for target.source) we could better use the target_record_id and the method that extracts the record id from the target

Reply:
Improved. I have removed a small code part in the SolrAnnotationUtils.processTargetUris that was separatelly adding the scope to the target ids, because the scope is already added in the extractUriValues, from which the target ids are made. So that now the target ids are extracted at one place, i.e. in the function SolrAnnotationUtils.extractRecordIds. Similarly I created a function SolrAnnotationUtils.extractMultilingualValue to capture setting of the multilingual body value at one place, because we use it also in the duplication check. Currently the value set is the same as for the body.value field (obtained through the SolrAnnotationUtils.extractTextValues), but maybe it can change in the future.